samtranslator/model/stepfunctions/events.py

```diff
@@ -161,11 +165,20 @@ def _construct_target(self, resource, role, dead_letter_queue_arn=None):  # type
             if self.Target and "Id" in self.Target
             else generate_valid_target_id(self.logical_id, EVENT_RULE_SFN_TARGET_SUFFIX)
         )
-        target = {
-            "Arn": resource.get_runtime_attr("arn"),
-            "Id": target_id,
-            "RoleArn": role.get_runtime_attr("arn"),
-        }
+
+        if provided_role:
+            role_arn = fnSub("arn:aws:iam::${AWS::AccountId}:role/${Role}", {"Role": provided_role})
```

Comment:
I see that [ScheduleV2 event](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-property-statemachine-statemachineschedulev2.html#sam-statemachine-statemachineschedulev2-rolearn) has a `RoleArn` property that is a straight pass through. Maybe we should do the same here to match that event + we don't have to worry about what the value is, we can just pass it through.